### PR TITLE
Reduce size of MANAGEMENT threadpool on small node

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today by default the `MANAGEMENT` threadpool always permits 5 threads
even if the node has a single CPU, which unfairly prioritises management
activities on small nodes. With this commit we limit the size of this
threadpool to the number of processors if less than 5.

Relates #70435

### CODE REVIEW

**Review by original-brownbear — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
Darn it. This change results in deadlocks in the `org.elasticsearch.http.*RestCancellationIT` tests for stats actions, because they rely on blocking the responding thread but they need another management thread for checking whether the tasks have started.

**Comment by DaveCTurner** [comment]
Darn it some more. `TransportCancelTasksAction` runs on the `MANAGEMENT` thread too. That seems like a bug, we might not be able to cancel remote management tasks if the remote node is too busy running them all. However it's not obvious we can move that to `SAME`, cancelling some tasks looks to involve nontrivial work.

**Comment by DaveCTurner** [comment]
@original-brownbear this has moved enough since your LGTM that it's worth another look.

@elasticmachine please test this (just for another CI run in case it shakes out anything else before merging)

**Comment by DaveCTurner** [comment]
once more with feeling ... @elasticmachine test this please

File: modules/reindex/src/test/java/org/elasticsearch/index/reindex/DeleteByQueryBasicTests.java
Comment (henningandersen) [inline modules/reindex/src/test/java/org/elasticsearch/index/reindex/DeleteByQueryBasicTests.java:244]: I like this simplification, so just out of curiosity, did not changing this cause a failure? I need a hint to realize how I think.

**Review by henningandersen — APPROVED** [review]
LGTM.

**Review by original-brownbear — APPROVED** [review]
LGTM2

File: modules/reindex/src/test/java/org/elasticsearch/index/reindex/DeleteByQueryBasicTests.java
Comment (DaveCTurner) [reply modules/reindex/src/test/java/org/elasticsearch/index/reindex/DeleteByQueryBasicTests.java:244]: Yes, the `threadPool.schedule(..., ThreadPool.Names.MANAGEMENT)` made a blocking call on the (sole) management thread that could not complete because the refresh needs to make stats calls which also need management threads. I did a targeted fix in b8a20ddd4cb84352ef947c5cd3e0378e6ffd5135 but then decided that the whole thing could be simplified.

Comment (henningandersen) [reply modules/reindex/src/test/java/org/elasticsearch/index/reindex/DeleteByQueryBasicTests.java:244]: Thanks, got it now.

**Pull request merged**
